### PR TITLE
feat(language-list): make the overlay knob work in 1 template - FRONT-491

### PR DESCRIPTION
**Pull request by Joosthe**
# PR description

Please drop a few lines about the PR: what it does, how to test it, etc.

## QA Checklist

In order to ensure a safe and quick review, please check that your PR follow those guidelines:

- [ ] I have put the vanilla component as `devDependencies`
- [ ] I have put the specs package as `devDependencies`
- [ ] I have added the components directly used in the twig file (with `include` or `embed`) as `dependencies`
- [ ] My component is listed in `@ecl-twig/ec-components`'s `dependencies`
- [ ] My variables naming follow the guidelines (snake case for twig)
- [ ] I have provided tests
- [ ] I have provided documentation (for the "notes" tab)
- [ ] If my local `yarn.lock` contains changes, I have committed it
- [ ] I have given my PR the proper label (`pr: review needed` to indicate that I'm done and now waiting for a review ,`pr: wip` to indicate that I'm actively working on it ...)

### CODE REVIEW

File: src/ec/packages/ec-component-language-list/language-list.story.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ import dataSplash from './demo/data--splash';
 import dataOverlay from './demo/data--overlay';
 
 import languageListSplash from './ecl-language-list-splash.html.twig';
```
Comment (papegaill) [inline src/ec/packages/ec-component-language-list/language-list.story.js]: if we use now one template maybe this _splash_ file needs to be renamed to `ecl-language-list-main.html.twig`?
`languageListSplash` can also be renamed to `languageList`?

File: src/ec/packages/ec-component-language-list/language-list.story.js [hunk 2]
```diff
@@ -23,7 +23,10 @@ storiesOf('Components/Language list', module)
     () =>
       languageListSplash(
         merge(dataSplash, {
-          icon_path: iconPath,
+          icon_path: text('Icon path', iconPath),
```
Comment (papegaill) [inline src/ec/packages/ec-component-language-list/language-list.story.js]: what the purpose of having `icon_path` as a knob?

File: src/ec/packages/ec-component-language-list/ecl-language-list-splash.html.twig [hunk 3]
```diff
@@ -22,8 +22,19 @@
 {% set _items = items|default([]) %}
 {% set _icon_path = icon_path|default('') %}
 {% set _logo = logo|default({}) %}
+{% set _overlay = overlay|default(false) %}
```
Comment (papegaill) [inline src/ec/packages/ec-component-language-list/ecl-language-list-splash.html.twig]: `overlay`, `close_label`and `title` should be added to parameters description at top of this file

File: src/ec/packages/ec-component-language-list/language-list.story.js (hunk 1 above)
Comment (papegaill) [reply src/ec/packages/ec-component-language-list/language-list.story.js]: you need also to remove `ecl-language-list-overlay.html.twig` file if it's no used anymore.

File: src/ec/packages/ec-component-language-list/ecl-language-list-splash.html.twig [hunk 4]
```diff
@@ -22,8 +22,19 @@
...
+{% set _close_label = close_label|default('') %}
+{% set _title = title|default('') %}
+
```
Comment (papegaill) [inline src/ec/packages/ec-component-language-list/ecl-language-list-splash.html.twig]: extra blank line to be removed

File: src/ec/packages/ec-component-language-list/ecl-language-list-splash.html.twig [hunk 5]
```diff
@@ -22,8 +22,19 @@
...
+
+{% if _overlay is defined and _overlay == true %}
+  {% set _css_class = 'ecl-language-list ecl-language-list--overlay' %}
+  {% set data_lang_overlay = 'data-ecl-language-list-overlay' %}
+{% else %}
+  {% set _css_class = 'ecl-language-list ecl-language-list--splash' %}
+{% endif %}
+
```
Comment (papegaill) [inline src/ec/packages/ec-component-language-list/ecl-language-list-splash.html.twig]: extra blank line to be removed

File: src/ec/packages/ec-component-language-list/ecl-language-list-splash.html.twig [hunk 6]
```diff
@@ -44,16 +55,55 @@
...
-    <img class="ecl-language-list__logo" src="{{ _logo.path }}" alt="{{ _logo.alt }}" />
-  </header>
+<div class="{{ _css_class }}" {{ _extra_attributes|raw }} {{ data_lang_overlay }}>
+  {% if _overlay is not defined or _overlay == false %}
+    <header class="ecl-language-list__header">
+      <img class="ecl-language-list__logo" src="{{ _logo.path }}" alt="{{ _logo.alt }}" />
+    </header>
+  {% endif %}
 
+
```
Comment (papegaill) [inline src/ec/packages/ec-component-language-list/ecl-language-list-splash.html.twig]: extra blank line to be removed

File: src/ec/packages/ec-component-language-list/ecl-language-list-splash.html.twig [hunk 7]
```diff
@@ -44,16 +55,55 @@
...
+          {{- _title -}}
+        </div>
+      </div>
+
+      {% include '@ecl-twig/ec-component-language-list/ecl-language-list.html.twig' with {
+        items: _items,
+        overlay: true,
+        icon_path: _icon_path
+      } only %}
+    {% else %}
```
Comment (papegaill) [inline src/ec/packages/ec-component-language-list/ecl-language-list-splash.html.twig]: add an extra blank line before `else` tag

File: src/ec/packages/ec-component-language-list/ecl-language-list-splash.html.twig (hunk 3 above)
Comment (papegaill) [reply src/ec/packages/ec-component-language-list/ecl-language-list-splash.html.twig]: It was already missing but `overlay` parameter should also be added on **Notes** tab parameters

File: src/ec/packages/ec-component-language-list/language-list.story.js (hunk 1 above)
Comment (Joosthe) [reply src/ec/packages/ec-component-language-list/language-list.story.js]: I left it like this for maybe older versions, who will still look for a twig template. But sure I will remove it.

File: src/ec/packages/ec-component-language-list/language-list.story.js (hunk 2 above)
Comment (Joosthe) [reply src/ec/packages/ec-component-language-list/language-list.story.js]: You can find this in the description of the ticket. To me this knob looks also a little bit to going far

Comment (planctus) [reply src/ec/packages/ec-component-language-list/language-list.story.js]: You are both right, ideally in my vision this should be a "fake" knob, a disabled text input showing the parameter and the value without giving the chance to modify it.
The reason why in this case this icon_path was mentioned as a knob is that it's usage is spread in the different component templates, but i agree that there are more disadvantages than advantages in having such a knob. Let's remove it then.

**Pull request merged**